tests/profiling/collector/test_threading.py

```diff
@@ -312,3 +313,40 @@ def test_lock_acquire_release_speed_patched(benchmark, pct):
 )
 def test_lock_acquire_release_speed(benchmark):
     benchmark(_lock_acquire_release, threading.Lock())
+
+
+@pytest.mark.skipif(not sys.platform.startswith("linux"), reason="only works on linux")
+@pytest.mark.subprocess(
+    ddtrace_run=True,
+    env=dict(DD_PROFILING_ENABLED="true"),
+    err=None,
+)
+def test_user_threads_have_native_id():
+    from os import getpid
+    from threading import Thread
+    from threading import _MainThread
+    from threading import current_thread
+    from time import sleep
+
+    main = current_thread()
+    assert isinstance(main, _MainThread)
+    # We expect the main thread to have the same ID as the PID
+    assert main.native_id == getpid(), (main.native_id, getpid())
+
+    t = Thread(target=lambda: None)
+    t.start()
+
+    for _ in range(10):
+        try:
+            # The TID should be higher than the PID, but not too high
```

Comment:
I think it's a good idea to sanity-test pid/tid integrity, but this assumption will break when
* PIDs are just about to wraparound
* The underlying PID namespace has a lot of activity, especially if the current task gets de-scheduled

Neither of these considerations is relevant for CI, so I think it's perfectly fine to take this section--just wanted to callout potential sources of flakiness.

Reply:
Indeed. I've thought about the first point you made and assumed it should be extremely rare to hit the wraparound boundary in CI. So hopefully we're good 🤞